File: generators/python/scripts/hand_joint_target.py

import ast

if joint_goal == 'joint_angles':
    hand_commander.move_to_joint_value_target_unsafe(ast.literal_eval(joint_pose_target), time_to_target, wait=wait, angle_degrees=angle)
else:
    joints = hand_commander.get_named_target_joint_values(joint_pose_target)
    hand_commander.move_to_joint_value_target_unsafe(joints, time_to_target, wait=wait, angle_degrees=angle)

File: sr_blockly_blocks/blocks/hand_joint_target.js
Blockly.Blocks['hand_joint_target'] = {
  init: function() {
    this.appendDummyInput()
        .appendField("Move hand to joint target:");
    this.appendDummyInput()
        .appendField(new Blockly.FieldDropdown([["Pose name", "enter_pose"], ["Enter angles", "joint_angles"]]), "joint_goal")
        .appendField(new Blockly.FieldTextInput("default"), "joint_pose_target");
    this.appendDummyInput()
        .appendField("Time:")
        .appendField(new Blockly.FieldTextInput("2"), "time_to_target")
        .appendField("Wait:")
        .appendField(new Blockly.FieldDropdown([["True", "True"], ["False", "False"]]), "wait")
        .appendField("Angle unit:")
        .appendField(new Blockly.FieldDropdown([["Radians", "False"], ["Degrees", "True"]]), "angle");
    this.setPreviousStatement(true);
    this.setNextStatement(true);
    this.setColour(240);
    this.setTooltip('Enter pose as either named pose or a list of joint angles in curly braces.');
    this.setHelpUrl('http://www.shadowrobot.com/');
  }
};
